### PR TITLE
ci: add ADR lint and PR validation

### LINKED ISSUE #677: Create a GitHub action to automate ADRs validation
Related to #663

### PR DESCRIPTION
**Pull request by Amzani**
**Description**

Fixes #677

### CODE REVIEW

File: .github/workflows/lint-pr-adr.yml
Comment (Amzani) [reply .github/workflows/lint-pr-adr.yml:26]: @KhudaDad414 I see your point and make sense, especially if we are planning to drop semantic release titles. However as the PR ADRs have no intent to trigger any release I'm wondering if we couldn't just create a new `ADR` repo for the AsyncAPI initiative and invite other projects to use it as well.

Any thoughts folks ?

Comment (fmvilas) [reply .github/workflows/lint-pr-adr.yml:26]: @KhudaDad414 what if instead of blocking this PR we make the regex to either support `chore: [ADR-NNNN]` or `[ADR-NNNN]` as the title? This way we can already unblock this, start using ADRs, and as soon as the migration to changesets is done, we can remove the `chore: [ADR-NNNN]` option.

Comment (Amzani) [reply .github/workflows/lint-pr-adr.yml:26]: I've seen this in the [doc](https://github.com/changesets/changesets/blob/main/docs/intro-to-using-changesets.md#some-handy-advice)
> Since changesets are focused on releases and changelogs, changes to your repository that don't require these won't need a changeset. As such, we recommend not adding a blocking element to contributions in the absence of a changeset.

It means that contributors will only commit the ADR files without the need to bump a release or use changesets ?

Comment (fmvilas) [reply .github/workflows/lint-pr-adr.yml:26]: Yeah, IMHO ADRs should not trigger a new release and, therefore, it shouldn't contain a changeset.

Comment (KhudaDad414) [reply .github/workflows/lint-pr-adr.yml:26]: @fmvilas agreed. let's merge this one.

Comment (KhudaDad414) [inline .github/workflows/lint-pr-adr.yml:26]:
```suggestion
            const re = /^(chore:\s*)?\[ADR-(\d{4})\]/;
```

File: .github/workflows/lint-pr-adr.yml
Comment (KhudaDad414) [inline .github/workflows/lint-pr-adr.yml:33]:
```suggestion
                message: "This PR Title must starts with `chore: [ADR-nnnn] name of ADR` or `[ADR-nnnn] name of ADR` where `nnnn` is the adr number (e.g `chore: [ADR-0001] use architecture decision records`) same us commits"
```

**Review by KhudaDad414 — APPROVED** [review]

**Comment by KhudaDad414** [comment]
@Amzani I have made the changes, please merge when ready. 🙇

**Pull request merged**
